Dockerfile

```diff
@@ -9,33 +9,22 @@ ENV OS_GIT_PATCH=${CI_Z_VERSION}
 ENV SOURCE_GIT_COMMIT=${CI_CLUSTER_LOGGING_OPERATOR_UPSTREAM_COMMIT}
 ENV SOURCE_GIT_URL=${CI_CLUSTER_LOGGING_OPERATOR_UPSTREAM_URL}
 ENV REMOTE_SOURCE=${REMOTE_SOURCE:-.}
+ENV REMOTE_SOURCE_DIR=${REMOTE_SOURCE_DIR:-/remote_source}
 
+WORKDIR $REMOTE_SOURCE_DIR/app
 
-WORKDIR /go/src/github.com/openshift/cluster-logging-operator
-
-COPY ${REMOTE_SOURCE}/main.go .
-COPY ${REMOTE_SOURCE}/apis ./apis
-COPY ${REMOTE_SOURCE}/controllers ./controllers
-COPY ${REMOTE_SOURCE}/internal ./internal
-COPY ${REMOTE_SOURCE}/must-gather ./must-gather
-COPY ${REMOTE_SOURCE}/version ./version
-COPY ${REMOTE_SOURCE}/scripts ./scripts
-COPY ${REMOTE_SOURCE}/files ./files
-COPY ${REMOTE_SOURCE}/go.mod .
-COPY ${REMOTE_SOURCE}/go.sum .
-COPY ${REMOTE_SOURCE}/manifests ./manifests
-COPY ${REMOTE_SOURCE}/.bingo .bingo
-COPY ${REMOTE_SOURCE}/Makefile ./Makefile
+COPY ${REMOTE_SOURCE} ${REMOTE_SOURCE}
+RUN go mod download
 
 USER 0
 RUN make build
 
-
-#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.8)
+#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.9)
 FROM quay.io/openshift/origin-cli:4.8 AS origincli
 
 #@follow_tag(registry.redhat.io/ubi8:latest)
 FROM registry.redhat.io/ubi8:8.5
+ENV REMOTE_SOURCE_DIR=${REMOTE_SOURCE_DIR:-/remote_source}
```

Comment:
We actually use `REMOTE_SOURCES` instead of `REMOTE_SOURCE_DIR`.

Reply:
This is the same as we do for MOLO and as advised in CPaaS users group. Ref:
* https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/containers_from_source_multistage_builds_in_osbs#jive_content_id_Cachito_Integration_for_gomod
* https://chat.google.com/room/AAAATiT7PR8/CqYbPOM1Ip8

This seems wrong and I'm surprised it works https://gitlab.cee.redhat.com/openshift-logging/cpaas-config/-/blob/openshift-logging-5.4-rhel-8/distgit/containers/cluster-logging-operator/Dockerfile.in#L11

Reply:
I'll update per our conversation with team